src/controller/gameController.ts

```diff
@@ -5,53 +5,74 @@ import { randomUUID } from 'crypto';
 
 import { LocalStorage } from '../LocalStorage/localStorage.js';
 import { em } from '../database.js';
-import { Game, Player, User } from '../entity/index.js';
-import { Card } from '../LocalStorage/Card.js';
-import { InviteResponse, InviteType } from '../LocalStorage/InviteResponse.js';
+import { ActionSchema, GameState, InviteResponseSchema, InviteType, PlayerSchema } from '../entity/schema.js';
+import { Action, Game, Player, User } from '../entity/index.js';
 import assert from 'assert';
 import { NopeSocket } from '../router/socketRouter.js';
+import { ref } from '@mikro-orm/core';
 
 export const gameInvite = errorWrapper(async (req: AuthRequest, res:Response) => {
-    // 1) Convert the input to a Game Objects
+    // check input
     if (!req.body.players || !Array.isArray(req.body.players)) {
         throw new AppError('Please provide an array of players', 400);
     }
     if (req.body.players.length < 2) {
         throw new AppError('Please provide at least two players', 400);
     }
+    const game = await initGame(req.body.noActionCards, req.body.noWildCards, req.body.oneMoreStartCard, req.body.players);
+    return res.status(200).send(game.toInvitationJSON());
+});
 
-    const game = new Game(randomUUID(), Card.getRandomCard());
+async function initGame(noActionCards: boolean, noWildCards: boolean, oneMoreStartCard: boolean, players: PlayerSchema[]) : Promise<Game> {
+    // 1) convert to game
+    const game = new Game(randomUUID(), noActionCards, noWildCards, oneMoreStartCard);
     const userRepository = em().getRepository(User);
-    for(let playerIndex=0, len = req.body.players.length; playerIndex < len; playerIndex++) {
-        const playerJson = req.body.players[playerIndex];
+    for(let playerIndex=0, len = players.length; playerIndex < len; playerIndex++) {
+        const playerJson = players[playerIndex];
         const user : User | null = await userRepository.findOne({'username':playerJson.username});
         if (user === null) {
             throw new AppError(`User ${playerJson.username} not found`, 400);
         }

```

Comment:
`playerJson` unnötig, da nur der `username` verwendet wird
```suggestion
        const username = players[playerIndex].username;
        const user : User | null = await userRepository.findOne({'username':username});
        if (user === null) {
            throw new AppError(`User ${username} not found`, 400);
        }
```